File: samples/AspireWithNode/AspireWithNode.AppHost/AppHost.cs
var builder = DistributedApplication.CreateBuilder(args);

var cache = builder.AddRedis("cache")
    .WithRedisInsight();

var weatherapi = builder.AddProject<Projects.AspireWithNode_AspNetCoreApi>("weatherapi")
    .WithHttpHealthCheck("/health");

builder.AddNodeApp("frontend", "../NodeFrontend", "./app.js")
    // .WithArgs(c =>
    // {
    //     c.Args.Insert(0, "--import");
    //     c.Args.Insert(1, "./instrumentation.js");
    // })
    .WithNpm()
    .WithRunScript("dev")
    .WithBuildScript("build")
    .WithHttpEndpoint(port: 5223, env: "PORT")
    .WithExternalHttpEndpoints()
    .WithHttpHealthCheck("/health")
    .WithReference(weatherapi).WaitFor(weatherapi)
    .WithReference(cache).WaitFor(cache);

builder.Build().Run();

File: samples/aspire-with-node/ts/apphost.ts
import { createBuilder } from './.modules/aspire.js';

const builder = await createBuilder();

const cache = await builder.addRedis("cache")
    .withRedisInsight();

const weatherapi = await builder.addProject("weatherapi", "../AspireWithNode.AspNetCoreApi/AspireWithNode.AspNetCoreApi.csproj", "https")
    .withHttpHealthCheck({
        path: "/health"
    });

await builder.addNodeApp("frontend", "../NodeFrontend", "./app.js")
    .withNpm()
    .withRunScript("dev")
    .withHttpEndpoint({ port: 5223, env: "PORT" })
    .withExternalHttpEndpoints()
    .withHttpHealthCheck({
        path: "/health"
    })
    .withServiceReference(weatherapi).waitFor(weatherapi)
    .withReference(cache).waitFor(cache);

await builder.build().run();
